backend.py

```diff
@@ -558,6 +558,8 @@ def partitionTargetDisk(disk, existing, preserve_first_partition, create_sr_part
     # Preserve any utility partitions unless user told us to zap 'em
     primary_part = 1
     if preserve_first_partition == 'true':
+        if 1 not in tool.partitionTable():  # If no first partition
```

Comment:
This method triggers the partition tool to rescan the disk which it has already done in the constructor. Generally this method should not be used publicly.

I think you want something like `if tool.getPartition(1):` instead.